src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -415,14 +422,11 @@ public override bool Equals(object? obj)
         /// </summary>
         public bool Equals(Half other)
         {
-            if (IsNaN(this) || IsNaN(other))
+            if (this == other)
```

Comment:
This makes the implementation shower than it needs to be - the `IsNaN` checks will be duplicated then the method returns false. 

I think it would be better to inline the logic here.

Reply:
I just copied the implementation from `double`, but didn't try to optimize it.
If there's more arithmetic operators for `Half` and corresponding benchmark, it will be better to track the optimization.

Reply:
`operator==` is very cheap for double. It is a method call for Half.

I think it is ok to merge this change as is into release/5.0. I have opened #41329 on the optimization.